src/groupeddataframe/splitapplycombine.jl

```diff
@@ -422,14 +422,27 @@ julia> combine(gd, :b => identity => :b, :c => identity => :c,
 
 See [`by`](@ref) for more examples.
 """
-combine(f::Union{Function, Pair}, gd::GroupedDataFrame; keepkeys::Bool=true) =
-    combine_helper(f, gd, keepkeys=keepkeys)
-# methods below need to be separate to avoid dispatch ambiguity
-combine(gd::GroupedDataFrame, f::Function; keepkeys::Bool=true) =
-    combine_helper(f, gd, keepkeys=keepkeys)
-combine(gd::GroupedDataFrame, f::Pair; keepkeys::Bool=true) =
+combine(f::Union{Base.Callable, Pair}, gd::GroupedDataFrame; keepkeys::Bool=true) =
+    combine(gd, f, keepkeys=keepkeys)
+combine(gd::GroupedDataFrame, f::Base.Callable; keepkeys::Bool=true) =
     combine_helper(f, gd, keepkeys=keepkeys)
 
+function combine(gd::GroupedDataFrame, f::Pair; keepkeys::Bool=true)
+    # move handling of aggregate to specialized combine
+    if isagg(first(f) => last(f) isa Pair ? first(last(f)) : last(f))
```

Comment:
```suggestion
    if isagg(first(f) => (last(f) isa Pair ? first(last(f)) : last(f)))
```

Reply:
OK. I will add it manually as I have too many changes locally.